graphql_compiler/compiler/emit_sql.py

```diff
@@ -104,6 +149,61 @@ def traverse(self, vertex_field, optional):
             onclause=(previous_alias.c[edge.from_column] == self._current_alias.c[edge.to_column]),
             isouter=optional)
 
+    def recurse(self, vertex_field, depth):
+        """Execute a Recurse Block."""
+        previous_alias = self._current_alias
+        edge = self._sql_schema_info.join_descriptors[self._current_classname][vertex_field]
+        primary_key = self._sql_schema_info.primary_keys[self._current_classname]
+        self._relocate(self._current_location.navigate_to_subpath(vertex_field))
+
+        if self._is_in_optional_scope():
+            raise NotImplementedError(u'The SQL backend does not implement mandatory '
```

Comment:
This error message is maybe not the best. Recurse is a mandatory traversal, but since it starts at depth 0 (i.e. the original node) it's always satisfiable by at least one path.

Let's reword this for clarity. I think the problem here is that `@recurse` inside `@optional` isn't supported, so let's say that. Keep in mind that when the user writes a big, complicated query that hits this error, this text is the only thing they'll see.